src/ofdm.c

```diff
@@ -1140,12 +1161,60 @@ static float est_timing_and_freq(struct OFDM *ofdm,
         complex float mvec[Npsam];
         for(int i=0; i<Npsam; i++) {
             complex float ph = cmplx(w*i);
-            mvec[i] = known_samples[i]*ph;
+            mvec[i] = conjf(known_samples[i]*ph);
         }
         for(int t=0; t<Ncorr; t+=tstep) {
+#if USE_VECTOR_OPS
+            float *rxPtr = (float*)&rx[t];
+            float *vecPtr = (float*)mvec;
+            float4 accumPos = { 0, 0, 0, 0 };
+            float4 accumNeg = { 0, 0, 0, 0 };
+            float4 accumImag = { 0, 0, 0, 0 };
+            float corrR = 0, corrI = 0;
+            int numBlocks = Npsam >> 1;
+            for (int i = 0; i < numBlocks; i++)
+            {
+                /* Lay out vectors as follows: 
+                   vec1 = rx[0].a, rx[0].b, rx[1].a, rx[1].b, ...
+                   vec2 = mvec[0].c, mvec[0].d, mvec[1].c, mvec1[1].d, ... */
+                float4 vec1 = { rxPtr[0], rxPtr[1], rxPtr[2], rxPtr[3] };
+                float4 vec2 = { vecPtr[0], vecPtr[1], vecPtr[2], vecPtr[3] };
+
+                accumPos += vec1 * vec2;
+                accumNeg -= vec1 * vec2;
+
+                /* Lay out vec3 as { rx[0].b, rx[0].a, rx[1].b, rx[0].b, ... }.
+                   Multiply vec3 by vec2 to get us bc, ad, bc, ad
+                   and add to second accumulator. */
+                float4 vec3 = { rxPtr[1], rxPtr[0], rxPtr[3], rxPtr[2] };
+                accumImag += vec3 * vec2;
+
+                /* Shift pointers forward by 4 (2 complex floats). */
+                rxPtr += 4; vecPtr += 4;
+            }
+
+            /* dot product: (a + bi)(c + di) = (ac - bd) + i(bc + ad) */
+            corrR = accumPos[0] + accumNeg[1] + accumPos[2] + accumNeg[3];
+            corrI = accumImag[0] + accumImag[1] + accumImag[2] + accumImag[3];
+            complex float corr = corrR + I * corrI;
+
+            /* Add remaining values to corr that couldn't be vectorized above. */
+            for (int i = numBlocks << 1; i < Npsam; i++)
+            {
+                corr += rx[i + t] * mvec[i];
+            }
+#elif __EMBEDDED__
+            float corrReal = 0, corrImag = 0;
+            codec2_complex_dot_product_f32((COMP*)&rx[t], (COMP*)mvec, Npsam, &corrReal, &corrImag);
```

Comment:
Looks like the current embedded platforms don't use this function, which makes sense since this only seems to deal with data modes. That said, I did find out that `codec2_complex_dot_product_f32` was implemented incorrectly on ezDV (I had been using `__REAL__`, so it wasn't getting used before).

Some benchmarks for 700D syncing on ezDV, BTW:

* gcc vector approach in this PR: ~160ms per 1280 samples
* ESP-DSP (see [here](https://github.com/tmiw/ezDV/commit/a3e9c461e4a42504e1a7406c2e133ff12ead595c)): ~61ms per 1280 samples
* `__REAL__` along with `__EMBEDDED__`: ~24ms per 1280 samples